app/models/lettings_log.rb

```diff
@@ -234,7 +234,13 @@ def applicable_income_range
 
     if hhmemb > 1
       (2..hhmemb).each do |person_index|
-        ecstat = self["ecstat#{person_index}"] || 10
+        ecstat = self["ecstat#{person_index}"]
+
+        if ecstat.nil?
+          age = self["age#{person_index}"]
+          # This should match the conditions under which ecstat is inferred as 9 (child under 16)
+          ecstat = age && age < 16 ? 9 : 10
+        end
```

Comment:
This did seem to be an easier (to write and to understand) solution than fiddling with the clearing logic, although it's slightly dissatisfying in that there could be similar problems elsewhere in future.